server/index.js

```diff
@@ -1,8 +1,23 @@
-
 import app from './src/configs/app';
+import database from './src/configs/database';
 
 const liveApp = app();
+const databaseUri = process.env.DB_URI;
 
-liveApp.create();
+const shutdownApplication = () => {
+  try {
+    database.closeDBConnections(() => {
+      liveApp.stop();
+    });
+  } catch (e) {
+    liveApp.stop();
+  }
+};
 
+liveApp.create();
 liveApp.start();
+
+database.connectDB(databaseUri);
+database.connectionError(shutdownApplication);
```

Comment:
This will throw error, if the database is not even connected.

Reply:
The callback will be executed only if the instance does not succeed in creating a connection with the database after the retries as well.